nengo_fpga/networks/fpga_pes_ensemble_network.py

```diff
@@ -143,18 +141,43 @@ def cleanup(self):
             os.remove(self.local_data_filepath)
 
     def connect_thread_func(self):
+        # Get the IP of the remote device from the fpga_config file
         remote_ip = fpga_config.get(self.fpga_name, 'ip')
 
+        # Get the SSH options from the fpga_config file
+        ssh_port = fpga_config.get(self.fpga_name, 'ssh_port')
+        ssh_user = fpga_config.get(self.fpga_name, 'ssh_user')
+
+        if fpga_config.has_option(self.fpga_name, 'ssh_pwd'):
+            ssh_pwd = fpga_config.get(self.fpga_name, 'ssh_pwd')
+        else:
+            ssh_pwd = None
+
+        if fpga_config.has_option(self.fpga_name, 'ssh_key'):
+            ssh_key = os.path.expanduser(fpga_config.get(self.fpga_name,
+                                                         'ssh_key'))
+        else:
+            ssh_key = None
+
         # Connect to remote location over ssh
-        self.ssh_client.connect(
-            fpga_config.get(self.fpga_name, 'ip'),
-            port=fpga_config.get(self.fpga_name, 'port'),
-            username=fpga_config.get(self.fpga_name, 'user'),
-            password=fpga_config.get(self.fpga_name, 'pwd'))
+        if ssh_key is not None:
```

Comment:
I think we should check `ssh_pwd` here first or change the default `fpga_config` not to have an entry for `ssh_key`. Basically I want this to work 'out of the box'.

Reply:
I put the `ssh_key` check first because the code should favour the use of an SSH key over a password (passwords are stored in plaintext, so will be less secure) in the event both are provided. I will remove the `ssh_key` from the `fpga_config` file and replace it with a comment.